discovery/azure/azure.go

```diff
@@ -188,59 +192,76 @@ func createAzureClient(cfg SDConfig) (azureClient, error) {
 		return azureClient{}, err
 	}
 
-	activeDirectoryEndpoint := env.ActiveDirectoryEndpoint
-	resourceManagerEndpoint := env.ResourceManagerEndpoint
-
 	var c azureClient
+	var credential azcore.TokenCredential
 
-	var spt *adal.ServicePrincipalToken
+	cloudConfig := cloud.Configuration{
+		ActiveDirectoryAuthorityHost: env.ActiveDirectoryEndpoint, Services: map[cloud.ServiceName]cloud.ServiceConfiguration{
+			cloud.ResourceManager: {
+				Audience: env.ServiceManagementEndpoint,
+				Endpoint: env.ResourceManagerEndpoint,
+			},
+		},
+	}
```

Comment:
Hi @tadelesh 

Shall we replace the `azure.EnvironmentFromName(...)` with a map for cloud.Configuration like this ?

```
var environments = map[string]Environment{
	"AZURECHINACLOUD":        cloud.AzureChina ,
	"AZUREGERMANCLOUD":       cloud.AzurePublic ,
	"AZURECLOUD":             cloud.AzurePublic ,
	"AZUREPUBLICCLOUD":       cloud.AzurePublic ,
	"AZUREUSGOVERNMENT":      cloud.AzureGovernment ,
	"AZUREUSGOVERNMENTCLOUD": cloud.AzureGovernment ,
}
```

Reply:
We do not use such environment any more in new SDK version. We have defined three common clouds' configuration in `azcore`. You have already used it: `cloud.AzureGovernment`, `cloud.AzureChina`, `cloud.AzurePublic`. I don't know whether you have exposed such environment settings to customer in your code, if so, then the mapping is fine to keep consistency, else, I think you could use cloud configuration directly.